.github/workflows/virusTotalAnalysis.js

```diff
@@ -1,70 +1,77 @@
-const glob = require('glob');
-const fs = require('fs');
-const { exec } = require('child_process');
+const glob = require("glob");
+const fs = require("fs");
+const { exec } = require("child_process");
+const countAPIUsageAndWait = require("./virusTotalAPISleepAndCount");
 
-function sleep(n) {
-  Atomics.wait(new Int32Array(new SharedArrayBuffer(4)), 0, 0, n);
-}
-
-function countAPIUsageAndWait({core}) {
-  // Sleep 20 seconds to avoid rate limiting
-  sleep(20 * 1000);
-  core._apiUsageCount++;
-}
 
 function writeVTScanUrl({core}, metadataFile, addonMetadata) {
-  // Write vtScanUrl to file
   const vtScanUrl = `https://www.virustotal.com/gui/file/${addonMetadata.sha256}`;
   addonMetadata.vtScanUrl = vtScanUrl;
   stringified = JSON.stringify(addonMetadata, null, "\t");
+  // Write vtScanUrl to add-on metadata file
   fs.writeFileSync(metadataFile, stringified);
   // Store the latest vtScanUrl for single file analysis
-  core.setOutput('vtScanUrl', vtScanUrl);
+  core.setOutput("vtScanUrl", vtScanUrl);
 }
 
+
 function getVirusTotalAnalysis({core}, addonMetadata, metadataFile, reviewedAddonsData) {
+  /*
+  Get the VirusTotal analysis for the add-on file.
+  If the add-on is flagged as malicious, store the sha256 hash in reviewedAddons.json.
+  Always store the scan URL in the add-on metadata file.
+  If Virus total fails to scan the add-on, fail the job.
+  */
   countAPIUsageAndWait({core});
   exec(`vt file ${addonMetadata.sha256} -k ${process.env.VT_API_KEY} --format json`, (err, stdout, stderr) => {
-    if (stderr !== '' || err !== null) {
+    if (stderr !== "" || err !== null) {
       console.log(`err: ${err}`);
       console.log(`stdout: ${stdout}`);
       console.log(`stderr: ${stderr}`);
-      core.setFailed('Failed to get VirusTotal analysis');
+      core.setFailed("Failed to get VirusTotal analysis");
       return;
     }
     writeVTScanUrl({core}, metadataFile, addonMetadata);
     // Append the VirusTotal analysis to the file for an artifact
     const vtData = JSON.parse(stdout);
-    fs.appendFileSync('vt.json', stdout);
+    fs.appendFileSync("vt.json", stdout);
     const stats = vtData[0]["last_analysis_stats"];
     const malicious = stats.malicious;
     if (malicious === 0) {
-      core.info('VirusTotal analysis succeeded');
+      core.info("VirusTotal analysis succeeded");
       return;
     }
     if (reviewedAddonsData[addonMetadata.addonId] === undefined) {
       reviewedAddonsData[addonMetadata.addonId] = [];
     }
     reviewedAddonsData[addonMetadata.addonId].push(addonMetadata.sha256);
     stringified = JSON.stringify(reviewedAddonsData, null, "\t");
-    fs.writeFileSync('reviewedAddons.json', stringified);
-    core.setFailed('VirusTotal analysis failed');
+    fs.writeFileSync("reviewedAddons.json", stringified);
+    core.setFailed("VirusTotal analysis failed");
```

Comment:
Is this supposed to be here? Seems like analysis hasn't failed, but the add-on has been flagged?

Reply:
This is so individual submissions will fail if they are flagged. We probably should have different behaviour for group runs vs individual runs